plugins/hls-explicit-record-fields-plugin/src/Ide/Plugin/ExplicitFields.hs

```diff
@@ -145,9 +154,9 @@ codeActionProvider ideState _ (CodeActionParams _ _ docId range _) = do
   let actions = map (mkCodeAction enabledExtensions) (RangeMap.filterByRange range crCodeActions)
   pure $ InL actions
   where
-    mkCodeAction :: [Extension] ->  Int -> Command |? CodeAction
+    mkCodeAction :: [Extension] -> Int -> Command |? CodeAction
     mkCodeAction exts uid = InR CodeAction
-      { _title = mkTitle exts
+      { _title = mkTitle exts -- TODO: `Expand positional record` without NamedFieldPuns if RecordInfoApp
```

Comment:
The existing `codeActionProvider` only needs "the list of extensions, and a int to allow us to resolve it later", but here `mkTitle` needs to be decided based on the `RecordInfo` how to generate it; If the `RecordInfo` is obtained here, does it not conform to the idea of ​​resolve?

should I adjust the `RecordInfo` to make two `codeActionProvider`? Or is there any better way?

Reply:
I think we probably don't want to get the `RecordInfo` here if we can avoid it, that's the expensive bit, right? 

I would have thought that the code action doesn't apply it all if it's a `RecordInfoApp`. Isn't the code action only for record wildcards?

Reply:
Yes, I also think we should avoid getting `RecordInfo` before resolve.

Since `RecordInfoApp` is a constructor of `RecordInfo`, so it is also processed in the code action provider that process `RecordInfo`.

When the code action resolves, we can get the `RecordInfo`, so a useless extension `RecordWildCards` will not actually be inserted if `RecordInfo` is `RecordInfoApp`.
The only problem with the current code is that the `title` of the code action will display **Expand record wildcard _(needs extension: NamedFieldPuns)_** (when the extension does not exist in current file), no matter what it is `RecordInfo`.

Reply:
I guess what I'm saying is that we shouldn't have any _code actions_ where the `RecordInfo` is `RecordInfoApp`, since `RecordInfoApp` can't be a wildcard, and that's what the code action applies to?